Spoony-iOS/Spoony-iOS/Resource/Components/SpoonyTextEditor.swift

```diff
@@ -0,0 +1,205 @@
+                    switch checkInputError(newValue) {
+                    case .maximumInputError(style: let style):
+                        errorState = .maximumInputError(style: style)
+                        text = String(newValue.prefix(style.maximumInput))
+                    case .minimumInputError(style: let style):
+                        errorState = .minimumInputError(style: style)
+                    case .noError:
+                        errorState = .noError
+                    }
+                }
+                .onChange(of: isFocused) { _, newValue in
+                    if !newValue, errorState.isMaximumInputError {
+                        errorState = .noError
+                    }
+                }
+
+            Text("\(text.count) / \(style.maximumInput)")
+                .font(.caption1m)
+                .foregroundStyle(errorState != .noError ? .error400 : .gray500)
+                .padding(.trailing, 5)
+                .padding(.bottom, 7)
+        }
+        .padding(.horizontal, 7)
+        .padding(.vertical, 5)
+        .frame(width: 335.adjusted, height: 125.adjustedH)
+        .background {
+            RoundedRectangle(cornerRadius: 8)
+                .stroke(borderColor, lineWidth: 1)
+        }
+    }
+    
+    // MARK: - errorMessageView
+    private func errorMessageView(_ message: String) -> some View {
+        HStack(spacing: 6) {
+            Image(.icErrorRed)
+                .resizable()
+                .frame(width: 16.adjusted, height: 16.adjusted)
+            
+            Text("\(message)")
+                .font(.caption1m)
+                .foregroundStyle(.error400)
+        }
+    }
+}
+
+// MARK: - Functions
+extension SpoonyTextEditor {
+    private func checkInputError(_ input: String) -> TextEditorErrorState {
+        let trimmedText = text.replacingOccurrences(of: " ", with: "")
+        
+        switch style {
+        case .review:
+            if input.count < style.minimumInput {
+                return .minimumInputError(style: .review)
+            } else if input.count > style.maximumInput {
+                return .maximumInputError(style: .review)
+            } else if errorState == .maximumInputError(style: .review),
+                      input.count == style.maximumInput {
+                return .maximumInputError(style: .review)
+            } else {
+                return .noError
+            }
+        case .report:
+            if trimmedText.count < style.minimumInput {
+                return .minimumInputError(style: .report)
+            } else if input.count > style.maximumInput {
+                return .maximumInputError(style: .report)
+            } else if errorState == .maximumInputError(style: .report),
+                      input.count == style.maximumInput {
+                return .maximumInputError(style: .report)
+            } else {
+                return .noError
+            }
+        }
+    }
+
```

Comment:
이 코드는 제가 봤을때, 공백이 있는지 없는지를 `switch` 문으로 해결하는 것 같습니다.
분기해야 할 것을 따로 상수로 뺀다면 중복 코드를 제거할 수 있을 것 같습니다!

개인적으로는 아래 처럼 분기하면 좋을 것 같습니다~

```suggestion
    private func checkInputError(_ input: String) -> TextEditorErrorState {
        let trimmedText = text.replacingOccurrences(of: " ", with: "")
        let currentInputLength = style == .report ? trimmedText.count : input.count

        if currentInputLength < style.minimumInput {
            return .minimumInputError(style: style)
        } else if currentInputLength > style.maximumInput {
            return .maximumInputError(style: style)
        } else if errorState == .maximumInputError(style: style),
                  currentInputLength == style.maximumInput {
            return .maximumInputError(style: style)
        } else {
            return .noError
        }
    }
```

Reply:
반영했습니다!